framework/wazuh/syscheck.py

```diff
@@ -148,64 +115,10 @@ def files(agent_id=None, summary=False, offset=0, limit=common.database_limit, s
-
-
-class WazuhDBQuerySyscheck(WazuhDBQuery):
-
-    def __init__(self, agent_id, summary, offset, limit, sort, search, select, query, count, get_data, default_sort_order='ASC', filters={}, min_select_fields=set()):
-
-        db_agent = glob('{0}/{1}-*.db'.format(common.database_path_agents, agent_id))
-        if not db_agent:
-            raise WazuhException(1600)
-        else:
-            db_agent = db_agent[0]
-
-        self.summary = False if summary == 'no' or not summary else True
-        if self.summary:
-            select = {'fields':["modificationDate", "event", "file", "scanDate"]} if not select else select
-        else:
-            select = {'fields':["scanDate", "modificationDate", "event", "file", "size", "octalMode", "user", "group", "md5", "sha1",
-                                "group", "inode", "gid", "uid"]} if not select else select
-
-        WazuhDBQuery.__init__(self, offset=offset, limit=limit, sort=sort, search=search, select=select, default_sort_field='date',
-                              query=query, db_path=db_agent, count=count, get_data=get_data, default_sort_order=default_sort_order,
-                              min_select_fields=min_select_fields, table='fim_event, fim_file', date_fields={'scanDate','modificationDate'},
-                              fields={'scanDate': 'date', 'modificationDate': 'mtime', 'file': 'path', 'size': 'size', 'user': 'uname',
-                                      'group': 'gname', 'event':'fim_event.type', 'md5':'md5', 'sha1':'sha1',
-                                      'inode':'inode','uid':'uid','gid':'gid', 'octalMode':'perm', 'filetype':'fim_file.type'},
-                              filters=filters)
-
-
-    def _default_query(self):
-        return "SELECT {0} FROM " + self.table + " WHERE fim_event.id_file = fim_file.id"
-
-
-    def _get_total_items(self):
-        if self.summary:
-            self.query += ' group by path'
-            self.conn.execute("SELECT COUNT(*) FROM ({0}) AS TEMP".format(self.query.format("max(date)")), self.request)
-            self.total_items = self.conn.fetch()[0]
-        else:
-            WazuhDBQuery._get_total_items(self)
-
-
-    def _get_data(self):
-        if self.summary:
-            self.fields['scanDate'] = 'max(date)'
-        WazuhDBQuery._get_data(self)
-
-
-    def _parse_legacy_filters(self):
-        if 'hash' in self.legacy_filters:
-            self.q += (';' if self.q else '') + '(md5={0},sha1={0})'.format(self.legacy_filters['hash'])
-            del self.legacy_filters['hash']
-        WazuhDBQuery._parse_legacy_filters(self)
+    select_fields = ["date", "mtime", "file", "size", "perm", "uname", "gname", "md5", "sha1", "sha256", "inode", "gid",
+                     "uid", "type"]
+    db_query = Agent(agent_id)._load_info_from_agent_db(table='fim_entry', select=select_fields, offset=offset,
+                                                        limit=limit, sort=sort, search=search, count=True)
```

Comment:
Search parameter isn't working. This is a key error. Are you using the correct structure for the `search` parameter?
```javascript
# curl -u foo:bar "localhost:55000/syscheck/000?pretty&search=privacy"
{
   "error": 1000,
   "message": "'fields'"
}
```